defaultconfig/template.conf

```diff
@@ -0,0 +1,196 @@
+    application/octet-stream iso img;
+    application/octet-stream msi msp msm;
+    application/json json;
+    audio/midi mid midi kar;
+    audio/mpeg mp3;
+    audio/ogg ogg;
+    audio/x-m4a m4a;
+    audio/x-realaudio ra;
+    video/3gpp 3gpp 3gp;
+    video/mp4 mp4;
+    video/mpeg mpeg mpg;
+    video/quicktime mov;
+    video/webm webm;
+    video/x-flv flv;
+    video/x-m4v m4v;
+    video/x-mng mng;
+    video/x-ms-asf asx asf;
+    video/x-ms-wmv wmv;
+    video/x-msvideo avi;
+  }
+
+  # TODO: Rename log format? Leave as is?
+  log_format paketo '$http_x_forwarded_for - $http_referer - [$time_local] "$request" $status $body_bytes_sent';
+  # TODO: write logs to /tmp?
+  access_log {{ env "APP_ROOT" }}/logs/access.log paketo;
+
+  # Set the default MIME type of responses; 'application/octet-stream'
+  # represents an arbitrary byte stream
+  default_type application/octet-stream;
+
+  # (Performance) When sending files, skip copying into buffer before sending.
+  sendfile on;
+  # (Only active with sendfile on) wait for packets to reach max size before
+  # sending.
+  tcp_nopush on;
+
+  # (Performance) Enable compressing responses
+  gzip on;
+  # For all clients
+  gzip_static always;
+  # Including responses to proxied requests
+  gzip_proxied any;
+  # For responses above a certain length
+  gzip_min_length 1100;
+  # That are one of the following MIME types
+  gzip_types text/plain text/css text/js text/xml text/javascript application/javascript application/x-javascript application/json application/xml application/xml+rss;
+  # Compress responses to a medium degree
+  gzip_comp_level 6;
+  # Using 16 buffers of 8k bytes each
+  gzip_buffers 16 8k;
+
+  # Add "Vary: Accept-Encoding” response header to compressed responses
+  gzip_vary on;
+
+  # Decompress responses if client doesn't support compressed
+  gunzip on;
+
+  # Don't compress responses if client is Internet Explorer 6
+  gzip_disable "msie6";
+
+  # Set a timeout during which a keep-alive client connection will stay open on
+  # the server side
+  # TODO: necessary? vv
+  keepalive_timeout 30;
+
+  # Ensure that redirects don't include the internal container PORT - <%=
+  # ENV["PORT"] %>
+  port_in_redirect off;
+
+  # (Security) Disable emitting nginx version on error pages and in the
+  # “Server” response header field
+  server_tokens off;
+
+  server {
+    listen {{port}};
+    # TODO: With only one server defined, all requests will be routed to this
+    # one even if they don't match this server name. Is this worth including,
+    # then?
+    server_name localhost;
+
+    # Directory where static files are located
+    root {{ env "APP_ROOT" }}/public;
+
+    location / {
+        # Specify files sent to client if specific file not requested (e.g.
+        # GET www.example.com/). NGINX sends first existing file in the list.
+        index index.html index.htm Default.htm;
+
+      # TODO: Allow users to include additional conf files?
```

Comment:
We could enable _extending_ the default config by allowing users to specify additional conf files to `include`. This adds a bit of scope to what's proposed in the zero-config [RFC](https://github.com/paketo-buildpacks/rfcs/blob/2803337e2cf88153535fa6df7ee7aecaad6e0b5d/text/web-servers/0004-zero-config.md). Is it still worth doing?